x/ibc-hooks/utils/utils.go

```diff
@@ -43,24 +44,23 @@ func MustExtractDenomFromPacketOnRecv(packet ibcexported.PacketI) string {
 	}
 
 	var denom string
-	if transfertypes.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel(), data.Denom) {
+	voucherPrefix := transfertypes.NewHop(packet.GetSourcePort(), packet.GetSourceChannel()).String()
+	if strings.HasPrefix(data.Denom, voucherPrefix) {
 		// remove prefix added by sender chain
-		voucherPrefix := transfertypes.GetDenomPrefix(packet.GetSourcePort(), packet.GetSourceChannel())
-
-		unprefixedDenom := data.Denom[len(voucherPrefix):]
+		unprefixedDenom := data.Denom[len(voucherPrefix)+1:]
```

Comment:
Is there a reason you need the intermediate variable here, or can you assign the value to `denom` directly?

Reply:
you're right